platform/ios/src/MGLMapView.mm

```diff
@@ -3187,6 +3187,11 @@ - (void)addAnnotations:(NS_ARRAY_OF(id <MGLAnnotation>) *)annotations
     {
         NSAssert([annotation conformsToProtocol:@protocol(MGLAnnotation)], @"annotation should conform to MGLAnnotation");
 
+        if ([annotation conformsToProtocol:@protocol(MGLFeature)])
+        {
+            NSLog(@"Warning: Annotations that conform to the MGLFeature protocol do not keep their `identifier` or `attributes` properties when added to the map as annotations. Add this feature using runtime styling if you intend to later query the map for it. %@", annotation.description);
```

Comment:
Let’s make this a one-time warning. It could get very spammy if you try to pass in a full array of features.